change/@ni-nimble-components-bcea55bf-ed59-4b11-b80b-6311749da460.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Adding filter input to Select",
```

Comment:
Moved to comment thread in-case discussion needed:
@rajsite @jattasNI @mollykreis, I wanted to call out explicitly new behaviors that I have introduced recently as part of this PR. I've already discussed this with both @mollykreis and @jattasNI and we all seem to be in agreement with the behavior. There are two notable new behaviors being introduced (and one minor one):

When a user navigates between options in the dropdown using the arrow keys (or by filtering) the display of the select will not update with the selected value in the dropdown (regardless of the filterMode setting). This helps avoid a confusing case for what a user should expect to happen after doing some navigation followed by filtering that removes all options from the dropdown. What should the user see in the display area in this case (particularly when it had been updating earlier as the user was navigating between options)? By not updating that display, this confusion is removed.
Pressing Esc while the dropdown is open when filter-mode is set to none will not commit the value. I imagine this behavior isn't controversial with the other behavior in place.
Both of these behaviors are actually called out explicitly in the [ARIA guidance for a 'Select-only' style combobx](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/), as noted departures from the native select input, but done because they improve the accessibility and usability of the component.

A minor new behavior I've introduced that I think is a slight usability improvement is preventing the Enter key from dismissing the popup if the user has filtered down to either only disabled options or no options at all. If a user wants to dismiss the popup in this state they can either press Esc or click away. Preventing the Enter key from dismissing is just a small hint to the user that they actually couldn't commit anything when perhaps that thought they could (perhaps they overtyped and filtered out an option they were meaning to filter to and quickly select).

Reply:
I think we should consider marking this a `major` change. While it doesn't break any APIs it is a behavior change which could theoretically break tests or corner case user behaviors.

We should also ensure we have Nimble tests for all the behaviors you describe in this comment (I haven't taken a look at the PR updates yet, just a reminder).

Reply:
One more corner case I noticed during testing (this is pre-existing behavior, not caused by this PR):
1. open select by clicking on it
2. dismiss popup via Enter or Esc
3. select still has keyboard focus so pressing Up/Down causes the value to change. However it doesn't have any visible focus indicator so this was slightly unexpected for me.

I leave it to you to decide what the correct behavior is in this case and whether we should implement that in this PR, file an issue, or treat it as an undefined corner case.

Reply:
My only non-pushy pushback on the idea that it is a major change, is that the only real functional change (in terms of programmatic observable state changes) is that we will no longer issue a change event when a user presses 'Esc'. If this is enough of a concern or grounds to make it a major change that is fine with me.

Reply:
I'll admit I'm on the fence and am willing to be outvoted, but my rationale for making it a major change is that it's plausible (though admittedly unlikely) that the behavior change could break a client test. I believe a client whose test was broken would appreciate the heads up and the cost on other clients of making it a major change isn't high.

Reply:
I think I can see it either way as well

> the only real functional change (in terms of programmatic observable state changes) is that we will no longer issue a change event when a user presses 'Esc'.

The same order of operations now has new meaning, open arrow, esc caused a value change and now open, arrow, esc doesn't cause a value change.

But I think ideally a test outside nimble shouldn't need to couple so highly to those operations. A page object should encapsulate a high level "select option x" and if that existed at that level a test wouldn't have broken. i.e. if the page object being added was already in use.

For an intended action we behave the same before and after. If the user changes an option an event is fired and if the user does not change an option and event is not fired. I think in this case I lean slightly towards not breaking and someone who breaks was overly coupled to implementation details (and we didn't give another choice really yet 😅) but I don't feel strongly.

Reply:
In the interest of resolving this relatively insignificant debate, I'm fine with `minor`.